### PR TITLE
feat: add more fields to rpc `get_peers`

### PR DESCRIPTION
**Pull request by quake**
added `connected_duration`, `last_ping_duration`, `protocols` and `sync_state` to `get_peers` rpc

### CODE REVIEW

File: rpc/README.md [hunk 1]
```diff
@@ -1908,9 +1918,52 @@ http://localhost:8114
...
+                    "id": "0x65",
+                    "version": "1"
+                },
+                {
+                    "id": "0x0",
+                    "version": "0.0.1"
+                }
+            ],
+            "sync_state": {
+                "best_known_header_hash": null,
```
Comment (driftluo) [inline rpc/README.md:1960]: Suggested add: `task_number`, it allows us to see the quality of service of the node during the synchronization period

Comment (quake) [reply rpc/README.md:1960]: Do you mean the `DownloadScheduler#task_count`?

Comment (driftluo) [reply rpc/README.md:1960]: yes

Comment (quake) [reply rpc/README.md:1960]: Got, will add it after https://github.com/nervosnetwork/ckb/pull/2190 is merged, need to resolve conflict first...

Comment (quake) [reply rpc/README.md:1971]: @driftluo added `can_fetch_count` / `inflight_count`, please review again

File: rpc/src/module/net.rs [hunk 2]
```diff
@@ -76,44 +79,102 @@ impl NetworkRpc for NetworkRpcImpl {
...
+                                .state()
+                                .read_inflight_blocks()
+                                .peer_inflight_count(*peer_index)
+                                as u64)
+                                .into(),
+                            can_fetch_count: (self
+                                .sync_shared
+                                .state()
+                                .read_inflight_blocks()
+                                .peer_inflight_count(*peer_index)
```
Comment (driftluo) [inline rpc/src/module/net.rs:161]: wrong function call, use `peer_can_fetch_count`

Comment (quake) [reply rpc/src/module/net.rs]: resolved, please review again, thanks

**Comment by quake** [comment]
bors r=doitian,driftluo

**Pull request merged**